### PR TITLE
Routing conventions code improvements

### PR DESCRIPTION
**Pull request by gathogojr**
Routing conventions code improvements

### CODE REVIEW

**Comment by gathogojr** [comment]
@xuzhg Please note that the change from `IsOperationParameterMeet` to `IsOperationParameterMatched` meets the definition of a breaking change. If you'd like us to defer the name correction please advise.

File: src/Microsoft.AspNetCore.OData/Routing/Conventions/ActionRoutingConvention.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public override bool AppliesToAction(ODataControllerActionContext context)
         }
 
         /// <inheritdoc />
-        protected override bool IsOperationParameterMeet(IEdmOperation operation, ActionModel action)
+        protected override bool IsOperationParameterMatched(IEdmOperation operation, ActionModel action)
```
Comment (xuzhg) [inline src/Microsoft.AspNetCore.OData/Routing/Conventions/ActionRoutingConvention.cs:50]: >IsOperationParameterMatched [](http://example.com/codeflow?start=32&length=27)

I agree to change it.

File: src/Microsoft.AspNetCore.OData/Routing/Conventions/NavigationRoutingConvention.cs [hunk 2]
```diff
@@ -57,15 +57,6 @@ public virtual bool AppliesToAction(ODataControllerActionContext context)
 
             ActionModel action = context.Action;
 
-            // Filter by the action name.
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Routing/Conventions/NavigationRoutingConvention.cs:60]: Moved this block of code a little further below. No use doing the splitting before other conditions are fulfilled, i.e.,
```csharp
            // filter by action parameter
            IEdmEntityType entityType = navigationSource.EntityType();
            bool hasKeyParameter = action.HasODataKeyParameter(entityType);
            if (!(context.Singleton != null ^ hasKeyParameter))
            {
                // Singleton, doesn't allow to query property with key
                // entityset, doesn't allow for non-key to query property
                return false;
            }
```

File: test/Microsoft.AspNetCore.OData.Tests/Routing/Conventions/RefRoutingConventionTests.cs [hunk 3]
```diff
@@ -95,7 +95,7 @@ public void SplitRefActionName_WorksAsExpected()
 
         [Theory]
         [MemberData(nameof(RefConventionTestData))]
-        public void SingletonRoutingConventionTestDataRunsAsExpected(string actionName, string[] templates)
+        public void RefRoutingConventionTestDataRunsAsExpected(string actionName, string[] templates)
```
Comment (xuzhg) [inline test/Microsoft.AspNetCore.OData.Tests/Routing/Conventions/RefRoutingConventionTests.cs:98]: >RefRoutingConventionTestDataRunsAsExpected [](http://example.com/codeflow?start=20&length=42)

Thanks for fixing it.

**Comment by xuzhg** [comment]
I think this change is ok.

---
In reply to: [884403334](https://github.com/OData/AspNetCoreOData/pull/245#issuecomment-884403334) [](http://example.com/codeflow?ancestors=884403334,884403334)

**Comment by xuzhg** [comment]
Looks good to me.

**Pull request merged**